src/lib/unlang/xlat.h

```diff
@@ -359,6 +359,8 @@ int		xlat_register_legacy(void *mod_inst, char const *name,
 
 xlat_t const	*xlat_register(TALLOC_CTX *ctx, char const *name, xlat_func_t func, bool needs_async);
 
+int xlat_func_arg(xlat_t const *xlat, xlat_input_type_t input_type, xlat_arg_parser_t args[]);
```

Comment:
I really think this needs to be two functions, one that takes an array of `xlat_arg_parser_t` and one that takes only a single one.  Else it's ambiguous what's meant by an args array when you have a MONO type input?